### ISSUE TITLE
[Frontend] CFG creation: 21445 (KeyError)

### ISSUE THREAD

**Issue by NeoQuix** [fkie-cad/dewolf#188]
### What happened?

```python
Traceback (most recent call last):
  File "/home/neoquix/Git-Repos/DeWolf/decompile.py", line 80, in <module>
    main(Decompiler)
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/util/commandline.py", line 87, in main
    task = decompiler.decompile(function_name, options)
  File "/home/neoquix/Git-Repos/DeWolf/decompile.py", line 54, in decompile
    task = self._frontend.create_task(function, task_options)
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/frontend/binaryninja/frontend.py", line 137, in create_task
    raise e
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/frontend/binaryninja/frontend.py", line 126, in create_task
    cfg = self._extract_cfg(function.function, options)
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/frontend/binaryninja/frontend.py", line 156, in _extract_cfg
    return parser.parse(function)
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/frontend/binaryninja/parser.py", line 37, in parse
    self._add_basic_block_edges(cfg, index_to_BasicBlock, basic_block)
  File "/home/neoquix/Git-Repos/DeWolf/decompiler/frontend/binaryninja/parser.py", line 50, in _add_basic_block_edges
    lookup_table[edge.target.source_block.start],
KeyError: 21445
```

### How to reproduce?

Decompile decode_format_string.part.0 from [od.zip](https://github.com/fkie-cad/dewolf/files/10868829/od.zip)

### Affected Binary Ninja Version(s)

Version 3.3.3996

**Comment by ebehner** [comment]
The problem is that the address of the jump-table is not in this function, instead it is an address of another function.
Consider the switch statement of the function:
![lifter](https://user-images.githubusercontent.com/7224664/234783759-15eabc33-6eec-41aa-ba86-89c3bd3ccea2.png)

To obtain the case-constants, we have a look at the lookup_table 
https://github.com/fkie-cad/dewolf/blob/e99a8050584d96f1b0d5f6fa051e205ed79d5855/decompiler/frontend/binaryninja/parser.py#L50
The lookup table has the following entries:
![look_up_table](https://user-images.githubusercontent.com/7224664/234784638-1b2f7c38-2fde-4bec-83f3-fe01323ac14e.png)
Now, the entry for case 0 belongs to the address `0x2800` and not the address `0x53c5` which is the first address of the leftmost block.
Instead, `0x2800` is the address of the block of function `decode_format_string.part.0.cold`:
![other_func](https://user-images.githubusercontent.com/7224664/234785067-556e95a3-36f9-41d1-9be5-898c21f31e13.png)

Not sure what we can do here. If we can not match a block to a case, we can have a look whether a very similar function is called there, and then look at this instruction and try to figure out whether this address matches, but this is neither clean nor nice:
![find_value](https://user-images.githubusercontent.com/7224664/234785923-ff387fb2-1fce-4ac4-bdbd-c334fc39a714.png)

**Comment by rihi** [comment]
Potentially fixed by #297. No longer able to reproduce.
